test/api/test_file_api.py

```diff
@@ -12,7 +12,9 @@ def test_get_files(flask_app):
         url_for("file_api.get_files")
     )
     assert response.status_code == 200
-    assert response.json == ["MY_DATASET", "OTHER_DATASET"]
+    assert response.json == [{'command': 'ADD_OR_CHANGE_DATA', 'datasetName': 'MY_DATASET'},
```

Comment:
Could it be that the testrunner in azure and your local environment returns the same list, but in a different order?
If that is the case it will pass if tested unordered:

```py
expected_response = [
    {'command': 'ADD_OR_CHANGE_DATA', 'datasetName': 'MY_DATASET'},
    {'command': 'PATCH_METADATA', 'datasetName': 'YOUR_DATASET'},
    {'command': 'ADD_OR_CHANGE_DATA', 'datasetName': 'OTHER_DATASET'}
]

for item in expected_repsonse:
    assert item in response.json
```